### PR TITLE
Add Leap approaches

### PR DESCRIPTION
**Pull request by glennj**
Largely based on the recent bash approaches PR.

### CODE REVIEW

File: exercises/practice/leap/.articles/performance/content.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+# Performance
+
+I'm demonstrating the use of [the `time` command][time-command] to compare the various approaches:
```
Comment (IsaacG) [inline exercises/practice/leap/.articles/performance/content.md:3]: Most articles aren't written in first person.

File: exercises/practice/leap/.articles/performance/content.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+1900 - 0 - 59.396 microseconds per iteration
+2000 - 1 - 72.496 microseconds per iteration```
+
+## Observations
+
+1. Boolean chain
+
+   We can see that the non leap year is the quickest to return, only having to execute one comparison.
+   The 100 and 400 years are the most expensive.
+   But still, we're talking under one microsecond execution time.
```
Comment (IsaacG) [inline exercises/practice/leap/.articles/performance/content.md:68]:
```suggestion
   Even 100 and 400, the most expensive years to compute, execute in under one microsecond.
```

**Review by IsaacG — APPROVED** [review]
I don't know Tcl but this all looks fine to me.

**Review by ErikSchierboom — APPROVED** [review]
Love it

**Pull request merged**